src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php

```diff
@@ -105,10 +105,11 @@ public function compilePrimary(Blueprint $blueprint, Fluent $command)
      */
     public function compileUnique(Blueprint $blueprint, Fluent $command)
     {
-        return sprintf('create unique index %s on %s (%s)',
+        return sprintf('create unique index %s on %s (%s) where %s is not null',
             $this->wrap($command->index),
             $this->wrapTable($blueprint),
-            $this->columnize($command->columns)
+            $this->columnize($command->columns),
+            $this->wrap($command->index)
```

Comment:
You have to insert the column name(s), not the index name.

Reply:
Yeh, I realized 😅 
I'll retarget, but it'll be something for next week as I have some more pressing issues & also some further local testing (telescope's migrations failed; I'll check asap)